.github/workflows/upgrade.yml

```diff
@@ -0,0 +1,128 @@
+name: Upgrade
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches:
+      - main
+      - "v[0-9]+.[0-9]+.[0-9]+"
+      - "v[0-9]+.[0-9]+"
+defaults:
+  run:
+    shell: bash
+    working-directory: metallboperator
+
+jobs:
+  main:
+    runs-on: ubuntu-20.04
+    env:
+      built_image: "metallb-operator:ci" # Arbitrary name
+    strategy:
+      matrix:
+        go: [ '1.16' ]
+    name: Go ${{ matrix.go }}
+    steps:
+    - name: Checkout Previous Release Metal LB Operator
+      uses: actions/checkout@v2
+      with:
+        path: metallboperator
+        ref: v0.12 # previous release version
+        fetch-depth: 0 # Fetch all history for all tags and branches
+
+    - uses: actions/setup-go@v2
+      id: go
+      with:
+        go-version: ${{ matrix.go }}
+
+    - name: Verify modules
```

Comment:
Just deploy and run the tests, we have other lanes for this kind of tests
Same for format, manifests and all the rest.

Reply:
changed it, hope its ok now.